lib/helpers/model/castBulkWrite.js

```diff
@@ -20,7 +20,7 @@ module.exports = function castBulkWrite(originalModel, op, options) {
       const model = decideModelByObject(originalModel, op['insertOne']['document']);
 
       const doc = new model(op['insertOne']['document']);
-      if (model.schema.options.timestamps != null) {
+      if (doc.initializeTimestamps) {
```

Comment:
Shouldn't this be `if (model.schema.options.timestamps)` ?

Reply:
I considered using it at first, but I tried to find where is also using `doc.initializeTimestamps()`, and I found this:

https://github.com/Automattic/mongoose/blob/9cf48f50d36791f43b517030c5c87a89964bcf75/lib/model.js#L3324-L3326

So I think it's better to use the same condition, if you need to change it in the future, it will be easier to find where you need to edit.
But it just my opinion, if you think it's better to use `if (model.schema.options.timestamps)`, I can change it.

Reply:
I think that is meant to handle a separate case. I'll make the changes on my own and merge this :+1: